mantidproject/mantid
# Speed up CalMuonDetectorPhases

Resolves #15066 

**Changes:** 
Uses James's suggestion to speed up the `CalMuonDetectorPhases` algorithm:
1. Calculate asymmetry using an appropriate grouping and fit this to find the frequency
2. Use the resulting frequency as the fixed, shared value of `w` in a subsequent sequential fit of all spectra to `f(x) = A*sin(w*x + p)`

The sequential fit is done using `PlotPeakByLogValue` (as used in the MDF interface), rather than duplicating this functionality. However, this required adding the input workspace to the ADS and removing it at the end.

The detector grouping is needed to calculate the asymmetry. New optional input properties allow the user to provide the forward and backward groups, but if they do not then the default grouping is read from the instrument definition. The code to do this already existed in the MuonAnalysis interface, so it has been refactored out into API in order to reuse it here. This doesn't work for MUSR because we don't know the field direction (this is in the NeXus file but not in the workspace), so for MUSR the user must supply the grouping.

Added the muon gyromagnetic ratio to `PhysicalConstants` as it was shared between this algorithm and `DynamicKuboToyabe`.

**To test:**
- Code review and verify tests pass
- Run `CalMuonDetectorPhases` on a muon dataset (for example `EMU00020711.nxs`) using both an installed version of Mantid and the version built from this PR. Verify that the results agree and also compare how long the algorithm took to run between the two versions.
- [x] [Release notes for 3.7](http://www.mantidproject.org/Release_Notes_3_7_MuonAnalysis#Algorithms) updated
